ui/window.go

```diff
@@ -885,6 +842,50 @@ func NewWindow(doRestart chan bool, app *tview.Application, session *discordgo.S
 	return window, nil
 }
 
+func (window *Window) TrySendMessage(message string) {
+	if window.selectedChannel == nil {
+		return
+	}
+
+	if len(message) == 0 {
+		if window.editingMessageID != nil {
+			msgIDCopy := *window.editingMessageID
+			window.askForMessageDeletion(msgIDCopy, true)
+		}
+		return
+	}
+
+	message = window.prepareMessage(message)
+	if len(message) > 2000 {
+		window.app.QueueUpdateDraw(func() {
+			window.ShowErrorDialog("Messages must be under 2000 characters to send")
+		})
+	} else if window.editingMessageID != nil {
+		go window.editMessage(window.selectedChannel.ID, *window.editingMessageID, message)
+		window.exitMessageEditMode()
+		window.OnMessageSent()
+	} else {
+		go func() {
+			messageText := window.jsEngine.OnMessageSend(message)
+			_, sendError := window.session.ChannelMessageSend(window.selectedChannel.ID, messageText)
+			window.chatView.internalTextView.ScrollToEnd()
+			if sendError == nil {
```

Comment:
Instead of having this block invoke `onMessageSent` or the error dialog, we could just wrap the whole if-block in a `window.app.QueueUpdateDraw(func() {` and remove the method `OnMessageSent`, as it is then only the clearing of text.

Reply:
Wrapping the entire statement in QueueUpdateDraw will cause rendering issues, because the drawing will often occur before the go-routine returns.